new and shiny

seanmonstar opened the pull request:
update all the things!

No but really, I was looking at our dependencies on gemnasium, and saw we had a lot out of date. We should stay uptodate, since they will include corner case bug fixes, and in some cases, speed improvements. This updates the majority of our dependencies, the ones that didn't really require much code changes to update.

I'll tackle mysql (0.9 to 2.0 promises 5x speed!), express, and folks in a later PR.

seanmonstar committed:
new and shiny

update all the things!

shane-tomlinson commented:
@seanmonstar - I had to add the following diff to be able to `npm install` after blowing away the `node_modules` directory:

``` diff
diff --git a/lockdown.json b/lockdown.json
index a64b58d..35d8b52 100644
--- a/lockdown.json
+++ b/lockdown.json
@@ -97,6 +97,9 @@
   "connect-cachify": {
     "0.0.15": "3a6d5329644e9303cff8a435c778732c5ed69396"
   },
+  "connect-cookie-session": {
+    "0.0.2": "cadd881d36e7d17fca102be77eded7f07846be1a"
+  },
   "connect-fonts": {
     "0.0.11": "805ccad0e46c89bcc9ebee47eb7f4254ce253d8b"
   },
```

Running tests now.

shane-tomlinson commented:
backend tests all pass, running Selenium tests.

shane-tomlinson commented:
@seanmonstar - With the updated deps, I am unable to push to a new ephemeral instance:

```
Yay!  You have your very own deployment.  Here's the basics:

 1. deploy your code:  git push update-everything HEAD:master
 2. visit your server on the web: https://update-everything.personatest.org
 3. ssh in with sudo: ssh ec2-user@update-everything.personatest.org
 4. ssh as the deployment user: ssh app@update-everything.personatest.org

 Here are your server's details: {
    "instanceId": "i-3512a779",
    "imageId": "ami-9e4da9e9",
    "instanceState": {
        "code": "16",
        "name": "running"
    },
    "dnsName": "ec2-54-220-241-190.eu-west-1.compute.amazonaws.com",
    "keyName": "awsbox deploy key (7f1323569463ac6cf4c726fb62084cfc)",
    "instanceType": "c1.medium",
    "ipAddress": "54.220.241.190",
    "launchTime": "2013-10-16T11:23:40.000Z",
    "tags": {},
    "name": "i-3512a779"
}
➜  ~PERSONA_ROOT git:(44a744c) ✗ git push update-everything HEAD:master
error: unable to push to unqualified destination: master
The destination refspec neither matches an existing ref on the remote nor
begins with refs/, and we are unable to guess a prefix based on the source ref.
error: failed to push some refs to 'app@54.220.241.190:git'
➜  ~PERSONA_ROOT git:(44a744c) ✗
```

seanmonstar commented:
Err, I don't get why my PRs run on my machine and Travis, but then you need
to update lockdown. Confused.

And dangit, I always forget about testing awsboxes.
On Oct 16, 2013 4:30 AM, "Shane Tomlinson" notifications@github.com wrote:

> @seanmonstar https://github.com/seanmonstar - With the updated deps, I
> am unable to push to a new ephemeral instance:
> 
> Yay!  You have your very own deployment.  Here's the basics:
> 1. deploy your code:  git push update-everything HEAD:master
> 2. visit your server on the web: https://update-everything.personatest.org
> 3. ssh in with sudo: ssh ec2-user@update-everything.personatest.org
> 4. ssh as the deployment user: ssh app@update-everything.personatest.org
>    
>    Here are your server's details: {
>      "instanceId": "i-3512a779",
>      "imageId": "ami-9e4da9e9",
>      "instanceState": {
>          "code": "16",
>          "name": "running"
>      },
>      "dnsName": "ec2-54-220-241-190.eu-west-1.compute.amazonaws.com",
>      "keyName": "awsbox deploy key (7f1323569463ac6cf4c726fb62084cfc)",
>      "instanceType": "c1.medium",
>      "ipAddress": "54.220.241.190",
>      "launchTime": "2013-10-16T11:23:40.000Z",
>      "tags": {},
>      "name": "i-3512a779"
>    }
>    ➜  ~PERSONA_ROOT git:(44a744c) ✗ git push update-everything HEAD:master
>    error: unable to push to unqualified destination: master
>    The destination refspec neither matches an existing ref on the remote nor
>    begins with refs/, and we are unable to guess a prefix based on the source ref.
>    error: failed to push some refs to 'app@54.220.241.190:git'
>    ➜  ~PERSONA_ROOT git:(44a744c) ✗
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/mozilla/browserid/pull/3981#issuecomment-26410427
> .

seanmonstar commented:
Oh right! I had removed connect-cookie-session in the other PR, so I musta dirtied my node_modules.

seanmonstar commented:
So, I've just built an awsbox also, and it's currently pushing. I wonder if your awsbox had a hiccup setting up. The dependencies shouldn't have affected that at all, since those aren't even sent to the box until you do your first git push.

shane-tomlinson commented:
@seanmonstar - can you rebase this?

seanmonstar commented:
rebased!

jaredhirsch commented on lockdown.json:
```diff
@@ -91,23 +92,23 @@
   "connect-cachify": {
-    "0.0.14": "7da8346c7b956ddb99fe248d1804bc2994132ec6"
-  },
-  "connect-cookie-session": {
```
@seanmonstar looks like rebasing did not restore this diff. i get the same lockdown error (connect-cookie-session missing) that shane was getting before.

jaredhirsch commented on lib/wsapi/session_context.js:
```diff
@@ -9,8 +9,7 @@ crypto = require('crypto'),
 secrets = require('../secrets.js'),
 version = require('../version.js'),
-config = require('../configuration.js'),
-Cookies = require('cookies');
```
oh, so this is the cookie-related change, i guess. do you mind removing this?

seanmonstar replied:
it's actually not. the connect-cookie-session change happened in https://github.com/mozilla/browserid/pull/3980 which is already merged. So, connect-cookie-session should correctly not be in the lockdown anymore.

jaredhirsch replied:
zomg, i am a goon today. thanks for clarifying

seanmonstar commented:
@6a68 wut? how? I rebased against dev, which no longer has connect-cookie-session at all. It works locally. Travis was passed all green.

Could it be cause I rebased and rewrote history? Did you use git pull and it couldn't pull?

jaredhirsch commented:
tests passing for me locally, json and mysql.

i am able to push to an awsbox without problems (well, provided i hack the connect-cookie missing dependency).

jaredhirsch commented on tests/jshint-test.js:
```diff
@@ -16,7 +16,7 @@ path = require('path'),
 
 var suite = vows.describe('jshint');
-var jshint = require('../node_modules/jshint/lib/hint').hint;
+var jshint = require('jshint').JSHINT;
```
this looks like random leftovers too?

seanmonstar replied:
nope, jshint's API changed since the previous version, so to update, this needed adjusting.

jaredhirsch replied:
o i c

jaredhirsch commented:
@seanmonstar idk what git magic has gone off the rails here, but I'd think the only files I should be seeing in this PR would be package.json and lockdown.json. /me shrugs.

Maybe amend the commit to remove the other changes, blow away node-modules, and see whether you can npm install cleanly?

jaredhirsch commented:
@seanmonstar now i am doubting my sanity. let me try again with a clean branch and node_modules all deleted and stuff.

jaredhirsch commented:
@seanmonstar in your branch, connect-cookie-session is actually listed inside package.json. Sup w/that?

jaredhirsch commented:
@seanmonstar oh, heh. my copy of your fork was out of date.

jaredhirsch commented:
updated fork has no problems, merging in.
